clang-tools-extra/clang-tidy/misc/CoroutineHostileRAIICheck.cpp

```diff
@@ -0,0 +1,82 @@
+//===--- CoroutineSuspensionHostileCheck.cpp - clang-tidy --------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "CoroutineHostileRAIICheck.h"
+#include "../utils/OptionsUtils.h"
+#include "clang/AST/Attrs.inc"
+#include "clang/AST/Decl.h"
+#include "clang/AST/ExprCXX.h"
+#include "clang/AST/Stmt.h"
+#include "clang/AST/Type.h"
+#include "clang/ASTMatchers/ASTMatchers.h"
+
+using namespace clang::ast_matchers;
+
+namespace clang::tidy::misc {
+CoroutineHostileRAIICheck::CoroutineHostileRAIICheck(StringRef Name,
+                                                     ClangTidyContext *Context)
+    : ClangTidyCheck(Name, Context) {
+  for (StringRef Denied :
+       utils::options::parseStringList(Options.get("RAIIDenyList", ""))) {
```

Comment:
1. Maybe all it RAIITypesList
2. Consider puting things like std::mutex by default here.

Reply:
> 1. Maybe all it RAIITypesList

Done.
> 2. Consider puting things like std::mutex by default here.

`std::mutex` is not a RAII type. `std::lock_guard`, `std::scoped_lock` are RAII and are already annotated with `scoped_lockable`

Reply:
scoped_lockable is an Clang language extension (https://releases.llvm.org/3.1/tools/clang/docs/LanguageExtensions.html), gcc std::lock_guard, std::scoped_lock are not mark with this attribute. So please put all those RAII objects on this list by default.

Reply:
Thanks. Makes sense then.